src/main/java/com/github/fge/jsonpatch/AddOperation.java

```diff
@@ -74,7 +76,7 @@
     public AddOperation(@JsonProperty("path") final JsonPointer path,
         @JsonProperty("value") final JsonNode value)
     {
-        super("add", path, value);
+        super(OPERATION_NAME, path, value);
```

Comment:
The way to resolve this is to have the factory pass the operation name into this constructor, so operation name is located only in the factory.  However, there are still some places where we instantiate the operations directly with the constructor by passing in path & value rather than from JsonNode (e.g. MoveOperation instantiates a RemoveOperation and an AddOperation), so these are places where we cannot use the factory in its current form.

In a future change, I think we can move instantiation by path & value into the factory itself, and highly discourage instantiation by the actual constructor.  What do you think?